[#7190] fix(catalogs): Validate and filter empty storageLocations in HadoopCatalogOperations

KyleLin0927 opened the pull request:
### What changes were proposed in this pull request?

Add validation: throw error if StorageLocation is null or blank  `Storage location for name '" + e.getKey() + "' must not be null or blank`

### Why are the changes needed?

Fixes #7190

### Does this PR introduce any user-facing change?

No. APIs and behavior are unchanged, only error messages for invalid input are clearer.

### How was this patch tested?

- Added new unit tests.
- Verified with `./gradlew clean build`.

mchades commented on catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/HadoopCatalogOperations.java:
```diff
@@ -245,6 +245,9 @@ public Fileset createMultipleLocationFileset(
       Map<String, String> properties)
       throws NoSuchSchemaException, FilesetAlreadyExistsException {
+    // remove the entries with empty value and check the default-location-names
+    storageLocations = filterAndValidateStorageLocations(storageLocations, properties);
```
We don't need a new method; we just need to add a validation line in line 301.

KyleLin0927 replied:
thanks for review.

KyleLin0927 commented:
@mchades  I have simplified the logic. throwing an exception if any blank data is found. I’ve also updated my unit tests accordingly and reverted the changes made to the web UI (no longer needed).

KyleLin0927 commented on catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/HadoopCatalogOperations.java:
```diff
@@ -296,6 +296,12 @@ public Fileset createMultipleLocationFileset(
+    }
     Map<String, Path> schemaPaths =
         getAndCheckSchemaPaths(schemaIdent.name(), schemaEntity.properties());
     if (schemaPaths.isEmpty() && storageLocations.isEmpty()) {
```
Thanks for suggestion. I applied this change, but during testing, I encountered a NullPointerException when the input storageLocations map contained a mix of valid values and blank/null values.

mchades replied:
NPE from where?

KyleLin0927 replied:
You may refer to the unit test I added in this PR (TestHadoopCatalogOperations.java line 1559)  If the storageLocations input contains both a valid entry and a blank value, it will still trigger an NPE in calculateFilesetPaths. If you think this test is not appropriate, I will remove it.

<img width="1400" alt="image" src="https://github.com/user-attachments/assets/04c2a29c-57ca-449f-b884-18d8d8f3b651" />

mchades replied:
let me think it over.

KyleLin0927 commented:
@mchades  Thansks for providing the code. I’ve completed the updates accordingly, but I noticed that the changes introduced caused the `testCreateSchemaWithEmptyCatalogLocation` fail. I made some adjustments to address the issue (you can find it in the second commit)

mchades commented on catalogs/catalog-hadoop/src/test/java/org/apache/gravitino/catalog/hadoop/TestHadoopCatalogOperations.java:
```diff
@@ -1493,6 +1494,24 @@ public void testCreateMultipleLocationsWithExceptions() throws IOException {
                       null));
       Assertions.assertEquals("Location name must not be blank", exception.getMessage());
 
+      // empty location in catalog location
```
could you plz also test the cases that:
1. empty location in the schema location
2. empty fileset storage location

mchades reviewed (approved):
LGTM. Thanks for your contributions!

Let's wait for the CI pass.
